### DISCUSSION TITLE
umi4插件register相同key——onGenerateFiles，如何保证最后一个执行

### DISCUSSION THREAD

**Discussion by wkeylin** [umijs/umi#9399] [Q&A] [answered]
需要在内置的tmpFiles plugin onGenerateFiles之后去做一些事情，所以自定义的插件也注册了onGenerateFiles，但是现有api无法保证内置所有onGenerateFiles执行完之后再执行自定义的onGenerateFiles

**Comment by wkeylin** [comment]
tmFiles 插件的onGenerateFiles 的stage 无穷大
<img width="720" alt="image" src="https://user-images.githubusercontent.com/21221921/191449465-19e24778-a88b-41b2-8161-04e5c56f012e.png">

**Comment by fz6m** [comment, marked as answer]
同样使用 `Infinity` ，后注册的会更晚执行。

```ts
  api.register({
    key: 'onGenerateFiles',
    fn: async () => {
      console.log('The last onGenerateFiles')
    },
    stage: Infinity,
  });
```
